rootex/core/resource_files/basic_material_resource_file.cpp

```diff
@@ -111,6 +111,16 @@ void BasicMaterialResourceFile::setLightmap(Ref<ImageResourceFile> lightmap)
 	m_MaterialData.lightmapImage = lightmap->getPath().generic_string();
 }
 
+Vector<Ref<Texture>> BasicMaterialResourceFile::getTextures() const
```

Comment:
make this a struct, Vectors can get confusing for such stuff
also integrate that struct into `MaterialResourceFile`